src/twisted/conch/ssh/transport.py

```diff
@@ -726,11 +726,18 @@ def dataReceived(self, data):
         if not self.gotVersion:
             if self.buf.find(b'\n', self.buf.find(b'SSH-')) == -1:
                 return
+
+            # RFC 4253 section 4.2 ask for strict `\r\n` line ending.
+            # Here we are a bit more relaxed and accept implementations ending
+            # only in '\n'.
+            # https://tools.ietf.org/html/rfc4253#section-4.2
```

Comment:
I'm always dubious of RFC violations but it's apparently pretty popular to end lines with just '\n'. [openssh even addresses this](https://github.com/openssh/openssh-portable/blob/b5974de1a1d419e316ffb6524b1b277dda2f3b49/ssh_api.c#L333-L341).

I appreciate you taking the time to comment the existing behavior!